### DISCUSSION TITLE
Break out translation importing into its own function for load{Locale,Namespace}Async

### DISCUSSION THREAD

**Discussion by jesperp** [ivanhofer/typesafe-i18n#384] [DX improvements]
Right now, `loadLocaleAsync` and `loadNamespaceAsync` imports translations _and_ loads them into the `loadedLocales` dictionary. I would like to break out the importing of the translation into a separate exported function that both `loadLocaleAsync` and `loadNamespaceAsync` can reuse. This opens up the possibility to get the translations without updating the globally shared `loadedLocales`, which can be problematic in a server environment ([see discussions here](https://github.com/ivanhofer/typesafe-i18n/discussions/315)).

This is untested possible solution (I am sure Ivan can clean this up):

```typescript
// New function to import translations
export const importTranslations = async <Namespace extends Namespaces>(
  locale: Locales,
  namespace?: Namespace
) => {
  const loader = namespace
    ? localeNamespaceLoaders[locale][namespace]
    : localeTranslationLoaders[locale];

  return (await loader()).default as unknown;
};

export const loadLocaleAsync = async (locale: Locales): Promise<void> => {
  updateDictionary(
    locale,
    //(await localeTranslationLoaders[locale]()).default as unknown as Translations  //Old
    await importTranslations(locale) as Translations // New
  );
  loadFormatters(locale);
};


export const loadNamespaceAsync = async <Namespace extends Namespaces>(
  locale: Locales,
  namespace: Namespace
): Promise<void> =>
  void updateDictionary(locale, {
    // Old:
    //[namespace]: (await (localeNamespaceLoaders[locale][namespace])()).default as unknown as Partial<Translations>
    [namespace]: await importTranslations(locale, namespace) as Partial<Translations>,  // New
  } as Partial<Translations>);

```

I think this is quite a simple structural change that doesn't break anything but opens up a possible solution for users that can't use the singleton dictionary. Sure, I can just do `"generateOnlyTypes": true` and make my own solution but then I loose a lot of the benefits like automatically updating my namespace imports. I think users wants to go great lengths to avoid generateOnlyTypes, unless doing a lot of custom stuff..

Thoughts? :)

**Comment by ivanhofer** [comment]
Thanks for your suggestion. Indeed it makes sense to expose a function to just load a dictionary. I can add this feature

**Reply by jesperp** [reply]
That would be awesome! I actually started fiddling around in the generator package in a new branch. Got a bit stuck since I wasn't familiar with a good workflow to update the failing tests. I guess you'd be a thousand times faster adding this feature but let me know if you're tight on time  🙏

**Reply by ivanhofer** [reply]
The snapshot tests are a bit tricky ^^

When you open a PR, you will see a description how to fix those tests.
I just realized that the description is missing inside the `CONTRIBUTING.md`. I'll add it also there.

Here is the description:

If you change something in the `generator`, please also:
 - add some snapshot tests to the `packages/generator/test/generator.test.ts`-file
 - then run `pnpm test` => you should see your tests failing
 - run `pnpm test:update-generated-files`
 - again run `pnpm test` => your tests should pass
 - then add all the created `*.expected.*`-files to the git-index and
 - make sure these files look like you would expect them
 - finally commit these files to the repository

Does this solve your issue?

**Reply by jesperp** [reply]
Thanks, I will give it a try!

**Comment by ivanhofer** [comment]
shipped in version `5.10.0`
